File: SDKs/googleapis.com/cloudtrace/v1/java/lib/src/main/java/openapisdk/models/shared/Trace.java
package openapisdk.models.shared;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonInclude.Include;

/**
 * Trace
 * A trace describes how long it takes for an application to perform an operation. It consists of a set of spans, each of which represent a single timed event within the operation.
**/
public class Trace {
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("projectId")
    public String projectId;
    public Trace withProjectId(String projectId) {
        this.projectId = projectId;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("spans")
    public TraceSpan[] spans;
    public Trace withSpans(TraceSpan[] spans) {
        this.spans = spans;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("traceId")
    public String traceId;
    public Trace withTraceId(String traceId) {
        this.traceId = traceId;
        return this;
    }
}

File: SDKs/googleapis.com/cloudtrace/v1/python/src/sdk/models/shared/trace.py
import dataclasses
from typing import Optional
from dataclasses_json import dataclass_json
from sdk import utils
from ..shared import tracespan as shared_tracespan


@dataclass_json
@dataclasses.dataclass
class Trace:
    r"""Trace
    A trace describes how long it takes for an application to perform an operation. It consists of a set of spans, each of which represent a single timed event within the operation.
    """
    
    project_id: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('projectId') }})
    spans: Optional[list[shared_tracespan.TraceSpan]] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('spans') }})
    trace_id: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('traceId') }})
    
